op-node/rollup/event/events.go

```diff
@@ -1,12 +1,21 @@
 package event
 
-import "github.com/ethereum/go-ethereum/log"
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/log"
+)
 
 type Event interface {
 	// String returns the name of the event.
 	// The name must be simple and identify the event type, not the event content.
 	// This name is used for metric-labeling.
 	String() string
+
+	// Context is attached to an event to provide meta-data about the source,
+	// the intention, and control any expiry (the event may become stale).
+	// Warning: this may return nil on some original event code that did not attach a context.
```

Comment:
I wonder if something central like the emitter should check if the context is nil and panic so we find these cases and can fix them. It seems like it would be very easy to miss setting the context and even if code handles the nil, that potentially loses important information from the context that should have been passed on to events emitted in response.

If we panic when an event with nil context is emitted even the most basic tests would catch it. Though the risk is that there's something very rare like an error event that slips though. It could be something obvious but less severe though like printing an error or even a stack trace.

Reply:
I have an alternative idea how we could enforce setting a context: https://github.com/ethereum-optimism/optimism/pull/16472#discussion_r2154128331